### ISSUE TITLE
Update entities by query

### ISSUE THREAD

**Issue by dortibi** [dotnet/efcore#23495]
I would like the ability for updating entity or entities by a query and not just by the entity PK.
Usages:

- I would like to update the entity only if the LastModifiedTime/Version isn't changed (I have the latest information)
- I have a few workers who trying to execute some action, before starting - each one of them trying to change the status from Pending -> InProgress, only one of then will succeed and start working, the rest will continue to the next action.
- I would like to update few entities without retrieving them to memory (without knowing their PKs), just call to SQL Update query.

A suggestions for usages:
must be a runtime protection that we able to translate the expressions  to SQL.

```
Class Entity
{
    public Guid Id { get; set; }
    public int Version { get; set; }
    public int IsEnable { get; set; }
    public string Name { get; set; }
}

// disable all old version
var rowsAffected = await dbContext.Set<Entity>().Where(e => e.Version == 0).UpdateAsync(e => e.IsEnable = false);

// update given entity only if i have latest version
Entity modifiedEntity;
var rowsAffected = await dbContext.Set<Entity>()
    .Where(e => e.Version == modifiedEntity.Version && e.Id == modifiedEntity.Id).UpdateAsync(e=> {
    e.Version += 1; 
    e.Name += modifiedEntity.Name; 
});

if(rowsAffected == 0)
{
    // I need to update my entity
}
```

**Comment by roji** [comment]
Duplicate of #795

**Comment by roji** [comment]
@dortibi please follow (and vote for) #795.
